tools/CouchDB_MRI_Importer.php

```diff
@@ -96,24 +96,82 @@ function _generateCandidatesQuery($ScanTypes)
+            // the subquery will return 'Multiple Selected files'. If no selected file //
+            // exists, the subselect will return 'No selected file'. Otherwise the     //
+            // file path of the selected file is returned by the subselect             //
+            //-------------------------------------------------------------------------//
+            $Query .= ', '
+                    . '('
+                    . ' CASE ('. $this->_getQueryForSelectedFiles('count(*)', $scantype) . ') '
+                    . '  WHEN 1 '
+                    . '    THEN (' . $this->_getQueryForSelectedFiles('f.File', $scantype) . ') '
+                    . '  WHEN 0 '
+                    . '    THEN "No selected file" '
+                    . '  ELSE "Multiple Selected files"'
+                    . ' END'
+                    . ") as Selected_$scantype";
+
+            //---------------------------------------------------------------------------//
+            // Add to the query a subselect that will compute the Qc status of the       //
+            // selected file for the given session/modality. If more than one selected   //
+            // file exists, the subquery will return 'Unknown: multiple Selected files'. //
+            // If no selected file exists, the subselect will return 'No selected file'. //
+            // Otherwise the subselect will return the Qc status of the selected file (or//
+            // 'No Qc on selected file' if the file has not been Qced)                   //
+            //---------------------------------------------------------------------------//
+            $Query  .= ', '
+                    . '('
+                    . ' CASE (' . $this->_getQueryForSelectedFiles('count(*)', $scantype) . ') '
+                    . '  WHEN 1 '
+                    . '    THEN ('
+                    .        $this->_getQueryForSelectedFiles(
+                                 'COALESCE(fqs.QCStatus, "No Qc on selected file")',
+                                 $scantype,
+                                 's.ID'
+                             )
+                    .     ') '
+                    . '  WHEN 0 '
+                    . '    THEN "No selected file" '
+                    . '  ELSE "Unknown: multiple Selected files"'
+                    . ' END'
+                    . ") as {$scantype}_QCStatus";
         }
-        $Query .= " FROM session s JOIN candidate c USING (CandID)
-            LEFT JOIN feedback_mri_comments fmric
-            ON (fmric.CommentTypeID=7 AND fmric.SessionID=s.ID)
-            WHERE c.Entity_type != 'Scanner' AND c.PSCID NOT LIKE '%9999'
-                  AND c.Active='Y' AND s.Active='Y' AND s.CenterID <> 1";
+
+        $Query .= " FROM session s"
+                . " JOIN candidate c USING (CandID)"
+                . " LEFT JOIN feedback_mri_comments fmric"
+                . " ON (fmric.CommentTypeID=7 AND fmric.SessionID=s.ID)"
```

Comment:
I'd define a constant here instead of using `7` (to give a clue as to what type of comment we're looking for).

Reply:
what type of comment are you looking for ?
 7 doesnt match many entries in RB

Reply:
I don't know about RB, but in my data I have:

```
mysql> select * from feedback_mri_comment_types WHERE CommentTypeID=7\G
*************************** 1. row ***************************
     CommentTypeID: 7
       CommentName: Subject
       CommentType: visit
CommentStatusField: 
1 row in set (0.00 sec)
```

Reply:
Don't look in RB, look in LORIS' `0000-00-00-schema.sql`. 7 Matches a comment type that can be applied to all the images acquired during the visit (e.g subject super stressed out to be in the scanner, so movement can be seen in all images, etc...)